### ISSUE TITLE
[Document] Remove development snapshot toolchain requirement in "Get Started" document

### ISSUE THREAD

**Issue by rayx** [apple/swift-testing#66]
### Description

The "Get Started" document says "A recent development snapshot toolchain is required to use the testing library". Is that still relevant after Swift 5.9 is released? This confused me when I first tried the library, but my experiment shows the library works fine with Xcode 15.0 on macOS.

**Comment by grynspan** [comment]
Yes, this is still relevant. As this package remains experimental, we expect to need to rely on language features and enhancements that are not available in 5.9.
